st2common/st2common/services/workflows.py

```diff
@@ -541,9 +541,12 @@ def request_action_execution(wf_ex_db, task_ex_db, st2_ctx, ac_ex_req, delay=Non
     # Identify the runner for the action.
     runner_type_db = action_utils.get_runnertype_by_name(action_db.runner_type['name'])
 
+    # Identify action pack name
+    pack_name = action_ref.split('.')[0] if action_ref else st2_ctx.get('pack')
+
     # Set context for the action execution.
     ac_ex_ctx = {
-        'pack': st2_ctx.get('pack'),
+        'pack': pack_name,
```

Comment:
Please default to st2_ctx.get('pack') when task is action less.

Reply:
` pack_name = action_ref.split('.')[0] if action_ref else st2_ctx.get('pack')`
With above line,  for action less, pack will be default to `st2_ctx.get('pack')` , not sure what need to be changed here. Please give more information.

Reply:
We're good here then.  See comment above. Thanks for following up.